File: python/prob_0003.py
'''
    Problem 3:
    Compute the largest prime factor of 600851475143

    Solved: 2014-08-21
'''


from PrimeTools import PrimeTools


def solve_prob(n=600851475143, pt=None):
    if pt is None:
        pt = PrimeTools(10**5)

    factorization = pt.factorize(n)
    prime_factors = [p for (p, e) in factorization]

    return max(prime_factors)


if __name__ == '__main__':
    soln = solve_prob()
    print("Solution to Prob 8: {}".format(soln))

File: haskell/prob_0003.hs
{-
    Problem 3:
    Compute the largest prime factor of 600851475143

    Solved: 2021-02-24
-}

-- https://hackage.haskell.org/package/primes
import Data.Numbers.Primes (primeFactors)

solveProb :: Integer
solveProb = maximum (primeFactors 600851475143)

main :: IO ()
main = do
    putStrLn $ "Solution to Prob 3: " ++ show solveProb
